lib/teiserver_web/controllers/account/security_controller.ex

```diff
@@ -68,4 +74,19 @@ defmodule TeiserverWeb.Account.SecurityController do
     |> put_flash(:info, "Token deleted successfully.")
     |> redirect(to: Routes.ts_account_security_path(conn, :index))
   end
+
+  @spec revoke_oauth_application(Plug.Conn.t(), map()) :: Plug.Conn.t()
+  def revoke_oauth_application(conn, %{"id" => application_id}) do
+    case User.revoke_application_access(conn.assigns.current_user.id, application_id) do
+      :ok ->
+        conn
```

Comment:
I would also log at `info` level which user revoked which application access there. Since it's a hard delete on the DB, it may be useful if we ever need to investigate.